adapters/repos/db/aggregator/filtered.go

```diff
@@ -135,38 +135,111 @@ func (fa *filteredAggregator) analyzeObject(ctx context.Context,
 func (fa *filteredAggregator) addPropValue(prop propAgg, value interface{}) error {
 	switch prop.aggType {
 	case aggregation.PropertyTypeBoolean:
-		asBool, ok := value.(bool)
-		if !ok {
-			return fmt.Errorf("expected property type boolean, received %T", value)
+		analyzeBool := func(value interface{}) error {
+			asBool, ok := value.(bool)
+			if !ok {
+				return fmt.Errorf("expected property type boolean, received %T", value)
+			}
+			if err := prop.boolAgg.AddBool(asBool); err != nil {
+				return err
+			}
+			return nil
 		}
-		if err := prop.boolAgg.AddBool(asBool); err != nil {
-			return err
+		switch prop.dataType {
+		case schema.DataTypeBoolean:
+			if err := analyzeBool(value); err != nil {
+				return err
+			}
+		case schema.DataTypeBooleanArray:
+			valueStruct := value.([]interface{})
```

Comment:
here you can also double check if the casting worked properly like so:
```
valueStruct, ok := value.([]interface{})
if !ok {
	return some error
}
```

Reply:
added, thanks!